microscopium/preprocess.py

```diff
@@ -631,8 +663,8 @@ def correct_multiimage_illumination(im_fns, illum, stretch_quantile=0,
     for im in ims_pass2:
         corrected = im / illum
         rescaled = exposure.rescale_intensity(corrected, in_range=corr_range,
-                                              out_range=np.uint8)
-        out = np.round(rescaled).astype(np.uint8)
+                                              out_range=im0.dtype.type)
+        out = np.round(rescaled).astype(im0.dtype)
```

Comment:
That's an interesting question ;) The only memory problem I encountered was when I was playing with Haralick features in ``mahotas``, where the [docs](https://mahotas.readthedocs.io/en/latest/faq.html#i-ran-out-of-memory-computing-haralick-features-on-16-bit-images-is-it-not-supported) state that memory issues can be a problem with >8 bit images

I'd rather preserve as much of the original data as possible and only scale down the images where we absolutely have to.